grype/presenter/sarif/presenter.go

```diff
@@ -151,34 +151,43 @@ func (pres *Presenter) helpText(m match.Match, link string) *sarif.MultiformatMe
 
 // packagePath attempts to get the relative path of the package to the "scan root"
 func (pres *Presenter) packagePath(p pkg.Package) string {
-	inputPath := strings.TrimPrefix(pres.srcMetadata.Path, "./")
-	if inputPath == "." {
-		inputPath = ""
-	}
 	if len(p.Locations) > 0 {
-		location := p.Locations[0]
-		packagePath := location.RealPath
-		if location.VirtualPath != "" {
-			packagePath = location.VirtualPath
-		}
-		if pres.srcMetadata.Scheme == source.DirectoryScheme {
-			packagePath = fmt.Sprintf("%s/%s", inputPath, packagePath)
+		return locationPath(p.Locations[0])
+	}
+	return pres.inputPath()
+}
+
+// inputPath returns a friendlier relative path or absolute path depending on the input, not prefixed by . or ./
+func (pres *Presenter) inputPath() string {
+	if pres.srcMetadata != nil {
+		inputPath := strings.TrimPrefix(pres.srcMetadata.Path, "./")
+		if inputPath == "." {
+			return ""
 		}
-		return packagePath
+		return inputPath
+	}
+	return ""
```

Comment:
nit: This impl could be less indented overall if the check for `pres.srcMetadata == nil` came first